src/components/Node.test.tsx

```diff
@@ -127,6 +129,47 @@ test("Load Note from other User which is not a contact", async () => {
   await screen.findByText("Bobs Note");
 });
 
+test("Load Connection types of non-contact user if necessary", async () => {
+  const [alice, bob] = setup([ALICE, BOB]);
+  const bobsDB = await setupTestDB(bob(), [["Bobs Note", []]]);
+  const node = findNodeByText(bobsDB, "Bobs Note") as KnowNode;
+
+  const planWithRelationTypes = planUpdateRelationTypes(
+    createPlan(bob()),
+    Map({
+      "1": { label: "Pro", color: "#FF0000" },
+      "": { label: "Default", color: "#000000" },
+    })
+  );
+  const plan = planUpsertRelations(
+    planWithRelationTypes,
+    newRelations(node.id, "1", bob().user.publicKey)
+  );
+  await execute({
+    ...bob(),
+    plan,
+    finalizeEvent: mockFinalizeEvent(),
```

Comment:
you don't need the finalizeEvent here, since it's in createPlan(bob()), so you can delete this line